ovs/extensions/healthcheck/volumedriver/volumedriver_health_check.py

```diff
@@ -161,60 +175,56 @@ def check_volumedrivers(logger):
                     if vp.guid in VolumedriverHealthCheck.MACHINE_DETAILS.vpools_guids:
                         try:
                             # delete if previous vdisk with this name exists
-                            if VolumedriverHealthCheck._check_volumedriver_remove(vpool_name=vp.name, vdisk_name=name,
-                                                                                  present=False):
-                                storagedriver_guid = next((storagedriver.guid for storagedriver in vp.storagedrivers
-                                                           if storagedriver.storagedriver_id == vp.name +
-                                                           VolumedriverHealthCheck.MACHINE_ID))
-                                # create a new one
+                            storagedriver_guid = next((storagedriver.guid for storagedriver in vp.storagedrivers
+                                                       if storagedriver.storagedriver_id == vp.name +
```

Comment:
If we ever change the naming convention of the storagedriver_ids, this will no longer match
Imo, its better to check for the matching StorageRouter

```
next(storagedriver.guid for storagedriver in vp.storagedrivers if storagedriver.storagerouter == VolumedriverHealthCheck.MACHINE_DETAILS)
```

PS: ```MACHINE_DETAILS``` is a very badly chosen name